packages/cli/src/s3-store-wrapper.ts

```diff
@@ -0,0 +1,101 @@
+import { StoreWrapperInterface, StoreSearchParams } from '@ceramicnetwork/core'
+import LevelUp from 'levelup'
+import S3LevelDOWN from 's3leveldown'
+import toArray from 'stream-to-array'
+import PQueue from 'p-queue'
+
+/**
+ * Maximum GET/HEAD requests per second to AWS S3
+ */
+const MAX_LOAD_RPS = 4000
+
+export class S3StoreWrapper implements StoreWrapperInterface {
```

Comment:
Mind taking a shot and calling it `IStoreWrapper`?

Reply:
Or, rather, see [comment ](https://github.com/ceramicnetwork/js-ceramic/pull/2557#discussion_r1024083401)